### PR TITLE
Remove `assert-news` utility

### PR DESCRIPTION
**Pull request by urutva**
### Description
`towncrier.check` supports checking news files in the current PR. Therefore we don't need to maintain `assert-news` custom script anymore.

### Test Coverage

- [x]  This change is covered by existing or additional automated tests.
- [ ]  Manual testing has been performed (and evidence provided) as automated testing was not feasible.
- [ ]  Additional tests are not required for this change (e.g. documentation update).

### CODE REVIEW

File: azure-pipelines/steps/install-development-dependencies.yml [hunk 1]
```diff
@@ -12,5 +12,6 @@ steps:
       python -m pipenv lock --dev -r --pre > dev-requirements.txt
       pip install -r dev-requirements.txt
       pip install pytest-azurepipelines
+      pip install towncrier
```
Comment (Patater) [inline azure-pipelines/steps/install-development-dependencies.yml]: Should towncrier be in `dev-requirements.txt` instead?

File: azure-pipelines/build-release.yml [hunk 2]
```diff
@@ -84,6 +84,7 @@ stages:
 
           # Static analysis is different for Python versions so should be run in each environment.
           - script: |
+              flake8 --version
```
Comment (Patater) [inline azure-pipelines/build-release.yml]: Adding `flake8 --version` doesn't have much to do with removing assert-news, so it should probably go in its own commit, if we want to add it.

Comment (urutva) [reply azure-pipelines/build-release.yml]: Good catch. I just added it for debugging. Removed now.

File: mbed_tools_ci_scripts/utils/git_helpers.py [hunk 3]
```diff
@@ -36,6 +36,11 @@ def __init__(self, path: Path, repo: Repo) -> None:
             configuration.get_value(ConfigurationVariable.BOT_EMAIL),
         )
 
+    def __del__(self) -> None:
```
Comment (rwalton-arm) [inline mbed_tools_ci_scripts/utils/git_helpers.py]: We don't claim to support python 3.5, do we need this fix for 3.6 or above?

Comment (rwalton-arm) [reply mbed_tools_ci_scripts/utils/git_helpers.py]: Also I would avoid using `__del__`, using it can break garbage collection in a number of ways. We'd be better off making this a context manager and calling `repo.close` in `__exit__`, as then at least the method is guaranteed to be called at the end of a `with` block.

File: azure-pipelines/steps/install-development-dependencies.yml (hunk 1 above)
Comment (urutva) [reply azure-pipelines/steps/install-development-dependencies.yml]: yeah moved `towncrier` required to `Pipfile`.

File: mbed_tools_ci_scripts/utils/git_helpers.py (hunk 3 above)
Comment (urutva) [reply mbed_tools_ci_scripts/utils/git_helpers.py]: > We don't claim to support python 3.5, do we need this fix for 3.6 or above?

Yeah CI failure is happening with Windows and python 3.6 not 3.5 (incorrect commit message).

Comment (urutva) [reply mbed_tools_ci_scripts/utils/git_helpers.py]: > Also I would avoid using `__del__`, using it can break garbage collection in a number of ways. We'd be better off making this a context manager and calling `repo.close` in `__exit__`, as then at least the method is guaranteed to be called at the end of a `with` block.

Ah k. Let me check that.

Comment (urutva) [reply mbed_tools_ci_scripts/utils/git_helpers.py]: The changes I did are based on what is done in gitpython https://github.com/gitpython-developers/GitPython/blob/60acfa5d8d454a7c968640a307772902d211f043/git/repo/base.py#L217. `gc.collect()` is invoked in `close()`. Is it ok to call garbage collection in `__exit__ `?

Comment (rwalton-arm) [reply mbed_tools_ci_scripts/utils/git_helpers.py]: Oh my word, that GitPython code is an awful hack. So many things wrong there. Anyway, we should be OK to call the `close` method in `__exit__`, they do it themselves https://github.com/gitpython-developers/GitPython/blob/60acfa5d8d454a7c968640a307772902d211f043/git/repo/base.py#L214 -- the garbage collector should only clean up any objects whose reference count has reached 0 (although there are other subtleties like generational garbage collection). I'm interested in what the test failure actually is and why we weren't seeing it before.

Comment (urutva) [reply mbed_tools_ci_scripts/utils/git_helpers.py]: CI failure only happens with Windows and python 3.6 https://dev.azure.com/mbed-tools/mbed-tools-ci-scripts/_build/results?buildId=3277&view=ms.vss-test-web.build-test-results-tab
`git.exc.GitCommandNotFound: Cmd('git') not found due to: OSError('[WinError 6] The handle is invalid')` is one of the errros.

**Review by Patater — APPROVED** [review]
LGTM

There may be still an open issue that @rwalton-arm and @urutva are still working through with the Windows git failure.

File: mbed_tools_ci_scripts/utils/git_helpers.py [hunk 4]
```diff
@@ -590,8 +590,15 @@ def __enter__(self) -> ProjectGitWrapper:
...
-        """Context manager exit point."""
-        pass
+        """Context manager exit point.
+
+        As described in
+        https://github.com/gitpython-developers/GitPython/blob/60acfa5d8d454a7c968640a307772902d211f043/git/repo/base.py#L223,
+        Tempfiles objects on Windows are holding references to open files until
+        they are collected by the garbage collector, thus preventing deletion.
+        """
+        print("__exit__ called")
```
Comment (rwalton-arm) [inline mbed_tools_ci_scripts/utils/git_helpers.py]: Should we remove this `print`?

Comment (rwalton-arm) [reply mbed_tools_ci_scripts/utils/git_helpers.py]: Also looks like we've got test failures. If your previous fix with `__del__` was working lets revert back to it, this will all be going away soon anyway.

Comment (urutva) [reply mbed_tools_ci_scripts/utils/git_helpers.py]: > Should we remove this `print`?

Removed, added for debugging purpose.

Comment (urutva) [reply mbed_tools_ci_scripts/utils/git_helpers.py]: > Also looks like we've got test failures. If your previous fix with `__del__` was working lets revert back to it, this will all be going away soon anyway.

One last attempt to use `__exit__` to fix the issue. If this doesn't work then I'll revert to `__del__` fix.

Comment (urutva) [reply mbed_tools_ci_scripts/utils/git_helpers.py]: Finally CI is happy with `__exit__` fix.

**Review by rwalton-arm — APPROVED** [review]
LGTM
